File: 144_Binary_Tree_Preorder_Traversal/144_1 (recursive).java
class Solution 
{
    private static void prepare(TreeNode root, List ans) 
    {
        if (root == null)
            return;
        ans.add(root.val);
        if (root.left != null)
            prepare(root.left, ans);
        if (root.right != null)
            prepare(root.right, ans);
    }

    public List<Integer> preorderTraversal(TreeNode root) 
    {
        List ans = new ArrayList();
        prepare(root, ans);
        return ans;
    }
}

File: 144_Binary_Tree_Preorder_Traversal/144_1 (recursive).py
class Solution:
    def preorderTraversal(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        if not root:
            return []
        return [root.val] +  self.preorderTraversal(root.left) + self.preorderTraversal(root.right)